### PR TITLE
Ccs 2706

### PR DESCRIPTION
**Pull request by benradey**
Significantly simplified module html in order to facilitate easy client-side sorting. Pretty CSS can be a future story.

### CODE REVIEW

File: src/main/java/com/redhat/pantheon/use/DateFormatter.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+@Model(adaptables = SlingHttpServletRequest.class)
+public class DateFormatter {
+
+    public String value;
+
+    @Inject
+    public DateFormatter(@Named("format") String format, @Named("date") Calendar date) {
+        value = new SimpleDateFormat(format).format(date.getTime());
+    }
+}
```
Comment (carlosmunoz) [inline src/main/java/com/redhat/pantheon/use/DateFormatter.java]: Wow, this I want to hear more about! :smile:

File: pom.xml [hunk 2]
```diff
@@ -57,7 +57,8 @@
                             SLING-INF/oak:index;overwrite:=true;uninstall:=true;path:=/oak:index
                         ]]></Sling-Initial-Content>
                         <Sling-Model-Packages>
-                            com.redhat.pantheon.model
+                            com.redhat.pantheon.model,
+                            com.redhat.pantheon.use
```
Comment (carlosmunoz) [inline pom.xml]: Mmm... lets talk about our package structure. We need to start thinking about how we want to structure our code... there are different options.

File: src/main/resources/apps/pantheon/modules/all/html.html [hunk 3]
```diff
@@ -10,35 +10,89 @@ <h1>All Modules</h1>
...
+        </sly>
+    </table>
 </div>
 
 <div data-sly-include="${'/apps/pantheon/partials/footer.html'}">
 </div>
+
+<!-- https://www.w3schools.com/howto/howto_js_sort_table.asp -->
+<script>
+function sortTable(n) {
```
Comment (carlosmunoz) [inline src/main/resources/apps/pantheon/modules/all/html.html]: This is the type of code we need to have unit tested and inlining it doesn't help with that. This is where a good front end framework helps (e.g. Angular, etc). We might need to discover how to do this while continuing to take advantage of HTL.

File: src/main/java/com/redhat/pantheon/use/DateFormatter.java (hunk 1 above)
Comment (benradey) [reply src/main/java/com/redhat/pantheon/use/DateFormatter.java]: What do you want to know? :)

This enables the passing of parameters via the data-sly-use property on src/main/resources/apps/pantheon/modules/all/html.html:29.

One note... Injection via constructor annotations works. Injection via field annotations, followed by a @PostConstruct method annotation, did not. No idea why. All I can say is that my favorite phrase "too much magic" applies here.

File: pom.xml (hunk 2 above)
Comment (benradey) [reply pom.xml]: Sure, open to suggestions. There is also a `<Sling-Model-Classes>` tag we can use to be more granular.

File: src/main/resources/apps/pantheon/modules/all/html.html (hunk 3 above)
Comment (benradey) [reply src/main/resources/apps/pantheon/modules/all/html.html]: Yeah, I knew this wasn't the greatest solution, but my thought was I'll get something quick and dirty in now, and we can implement a proper UI in a future story.

File: src/main/java/com/redhat/pantheon/use/DateFormatter.java (hunk 1 above)
Comment (benradey) [reply src/main/java/com/redhat/pantheon/use/DateFormatter.java]: IMO, sorting is a good candidate for being done client-side. The server doesn't have any info the client doesn't have and waiting for the round-trip takes time. That said, I could imagine at least one way to accomplish a server-side sort using URL params. (It's clumsy, and I'm sure there are better ways, but I don't know them offhand).

Comment (benradey) [reply src/main/java/com/redhat/pantheon/use/DateFormatter.java]: You know, when I had been thinking about sorting, it was sorting titles - and I had thought it was reasonable to always send the full list to the client. Now I realize we're talking about sorting modules, which could be much more numerous. In that case perhaps server-side sorting is the best approach.

**Pull request merged**